### DISCUSSION TITLE
Plan for MAUI on Windows - WinUI 2.x instead of WinUI 3.x?

### DISCUSSION THREAD

**Discussion by Aculeo** [dotnet/maui#1684] [General]
Now that [WinUI 3 for UWP has effectively been cancelled](https://www.theregister.com/2021/07/02/uwp_microsoft_winui3/) (it won't get a stable release), can you please elaborate on what the plan is for targeting Windows with MAUI? In general, UWP seems to have a rather uncertain future, that is at least my impression so far.  I wonder if we will have to eventually convert our applications back to Win32 instead of UWP.

**Comment by GeraudFabien** [comment]
[Maui](https://devblogs.microsoft.com/dotnet/announcing-net-maui-preview-5/#single-project-templates-updates)/[Uno](https://platform.uno/docs/articles/uwp-vs-winui3.html)/[React](https://microsoft.github.io/react-native-windows/docs/winui3) are already using WinUI3.
And WinUI3 still work (and only work) on windows. So if you target windows use WinUI3.

**Reply by Aculeo** [reply]
Thanks, I know they are using WinUI3 currently, which is not getting a stable release for UWP - so will MAUI switch to using Win32 + WinUI3, or UWP + WinUI2? I can't imagine they will release a stable version of MAUI with WinUI3 Preview as the target

**Reply by GeraudFabien** [reply]
I don't understand what the fact that WinUi3 hasn't uwp support change for maui use.  WinUI3 still work on windows so it doesn't change anythinks for it.

[WinUI3 has stable release](https://docs.microsoft.com/en-us/windows/apps/winui/winui3/#:~:text=WinUI%203%20-%20Project%20Reunion%200.5%20is%20the,production-ready.%20Install%20WinUI%203%20-%20Project%20Reunion%200.5)

> I can't imagine they will release a stable version of MAUI with WinUI3 Preview as the target

Well i can't answer that. But you doesn't have to target the first release if it didn't comply with you're need.

**Reply by Aculeo** [reply]
It matters because, so far, Xamarin.Forms + MAUI have always produced a UWP application. WinUI3 only has a stable release for Win32. If MAUI were to suddenly change direction to produce a Win32 executable, that would be a big change impacting deployment process, sandboxing no longer being a thing, access to UWP APIs, etc.
